components/plugin/src/miniboxing/plugin/metadata/MiniboxMetadataWarnings.scala

```diff
@@ -158,17 +158,20 @@ trait MiniboxMetadataWarnings {
 
   class UseMbArrayInsteadOfArrayWarning(p: Symbol, tpe: Type, pos: Position, inLibrary: Boolean = false) extends MiniboxWarning(p, pos, inLibrary) {
 
-    override def msg: String = "Use MbArray instead of Array and benefit from miniboxing specialization. " +
-															 "For more details about MbArrays, please check the following link: " +
-															 "http://scala-miniboxing.org/arrays.html"
+    override def msg: String = UseMbArrayInsteadOfArrayWarning.msg
 
     override def shouldWarn(): Boolean = {
       flags.flag_warn_mbarrays &&
-      p.owner.isArray &&
-      (ScalaValueClasses.contains(tpe.typeSymbol) || tpe.typeSymbol.deSkolemize.hasAnnotation(MinispecClass))
+      ((p.owner.isArray || (p == ArrayModule_genericApply)) && tpe.typeSymbol.deSkolemize.hasAnnotation(MinispecClass) || p.owner.isClassTag)
     }
   }
 
+  object UseMbArrayInsteadOfArrayWarning {
+    def msg: String = "Use MbArray instead of Array and benefit from miniboxing specialization. " +
+                      "For more details about MbArrays, please check the following link: " +
+                      "http://scala-miniboxing.org/arrays.html"
```

Comment:
How about:

```
   def msg: String = "Use MbArray instead of Array to eliminate the need for ClassTags and " +
                     "benefit from seamless interoperability with the miniboxing specialization. " +
                     "For more details about MbArrays, please check the following link: " +
                     "http://scala-miniboxing.org/arrays.html"
```